src/lock_file/satisfiability.rs

```diff
@@ -281,69 +340,32 @@ pub fn verify_platform_satisfiability(
 
 enum Dependency {
     Conda(MatchSpec, Cow<'static, str>),
-    PyPi(RequirementOrEditable, Cow<'static, str>),
+    PyPi(pypi_types::Requirement, Cow<'static, str>),
 }
 
 /// Check satatisfiability of a pypi requirement against a locked pypi package
 /// This also does an additional check for git urls when using direct url references
 pub fn pypi_satifisfies_editable(
+    spec: &pypi_types::Requirement,
     locked_data: &PypiPackageData,
-    spec: &EditableRequirement,
 ) -> bool {
-    let spec_url = &spec.url;
-
-    // In the case that both the spec and the locked data are direct git urls
-    // we need to compare the urls to see if they are the same
-    let spec_git_url = ParsedGitUrl::try_from(spec_url.to_url().clone()).ok();
-    let locked_git_url = locked_data
-        .url_or_path
-        .as_url()
-        .and_then(|url| ParsedGitUrl::try_from(url.clone()).ok());
-
-    // Both are git url's
-    if let (Some(spec_git_url), Some(locked_data_url)) = (spec_git_url, locked_git_url) {
-        let base_is_same = spec_git_url.url.repository() == locked_data_url.url.repository();
-
-        // If the spec does not specify a revision than any will do
-        // E.g `git.com/user/repo` is the same as `git.com/user/repo@adbdd`
-        if *spec_git_url.url.reference() == GitReference::DefaultBranch {
-            return base_is_same;
-        }
-
-        // If the spec has a short commit than we can do a partial match
-        // E.g `git.com/user/repo@adbdd` is the same as `git.com/user/repo@adbdd123`
-        // Currently this resolves to BranchOrTag
-        if let GitReference::BranchOrTag(branch_or_tag) = spec_git_url.url.reference() {
-            if seems_like_commit_sha(branch_or_tag) {
-                // We expect the lock file to have a long commit hash
-                // in this case
-                if let GitReference::FullCommit(sha) = locked_data_url.url.reference() {
-                    return base_is_same && sha.starts_with(branch_or_tag);
+    // We dont match on spec.is_editable() != locked_data.editable
+    // as it will happen later in verify_package_platform_satisfiability
+    // TODO: could be a potential refactoring opportunity
+
+    match &spec.source {
+        RequirementSource::Registry { .. } => false,
+        RequirementSource::Url { .. } => false,
+        RequirementSource::Git { .. } => false,
```

Comment:
Same here, does returning false mean we cant check these? Or are these never editable?

Reply:
we always store editables under `UrlOrPath::Path` variant , so we need to verify only the path

Reply:
Same, should we use unreachable! then?